apps/backend/api/src/modules/auth/auth.service.ts

```diff
@@ -95,24 +118,55 @@ export class AuthService {
         {
           success: false,
           message: 'Too many OTP requests. You are banned for 24 hours.',
-          error: { code: 'TOO_MANY_REQS' },
         },
         HttpStatus.BAD_REQUEST,
       );
     }
   }
 
-  // Store the generated OTP in Redis with a 5-minute validity
   private async storeOtp(phoneNumber: string, otp: string): Promise<void> {
     const otpKey = `otp:${phoneNumber}`;
     const otpRequestKey = `otp_request:${phoneNumber}`;
     const secret = this.otpService.generateSecret(otp);
 
-    await this.redisService.set(otpKey, secret, 'EX', 300); // 5 minutes validity
-    await this.redisService.set(otpRequestKey, 'requested', 'EX', 360); // Slightly longer TTL
+    await this.redisService.set(otpKey, secret, 'EX', 180);
+    await this.redisService.set(otpRequestKey, 'requested', 'EX', 360);
+  }
+
+  private async sendOtp(phoneNumber: string, otp: string): Promise<void> {
+    if (
+      !phoneNumber ||
+      typeof phoneNumber !== 'string' ||
+      phoneNumber.trim() === ''
+    ) {
```

Comment:
```suggestion
    if (
      !phoneNumber
    ) {
```

We must only check if the number is present or not. The type must be string; this can be handled from the submission making sure only string value is send. We should trim the value rather than throwing an error.